specification/ai/Face/routes.persondirectory.tsp

```diff
@@ -0,0 +1,254 @@
+  @doc(
+    """ 
+  * Dynamic person groups are stored in alphabetical order of dynamic person group id created in PersonDirectory DynamicPersonGroup - Create.
+  ${ListRequestOptionsDescription}
+  """
+  )
+  @get
+  @action("dynamicPersonGroupReferences")
+  @actionSeparator("/")
+  getDynamicPersonGroupReferences is Foundations.ResourceOperation<
+    PersonDirectoryPerson,
+    ListRequestOptions,
+    ListGroupReferenceResult,
+    ServiceTraits
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/byos" "It's an RPC call to detect face from an image, it doesn't store anything. There's a BYOS version of this call, but it's not this one."
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  #suppress "@azure-tools/typespec-azure-core/long-running-polling-operation-required" "Doesn't fit in generated long-running operation"
+  addPersonFace is AddPersonFaceOperation<{
+    @doc("The format of the HTTP payload.")
+    @header
+    contentType: "application/octet-stream";
+
+    @doc("The image to be analyzed")
+    @body
+    imageContent: bytes;
+  }>;
+
+  #suppress "@azure-tools/typespec-azure-core/byos" "It's an RPC call to detect face from an image, it doesn't store anything. There's a BYOS version of this call, but it's not this one."
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  #suppress "@azure-tools/typespec-azure-core/long-running-polling-operation-required" "Doesn't fit in generated long-running operation"
+  addPersonFaceFromUrl is AddPersonFaceOperation<{
+    @doc("URL of input image.")
+    url: url;
+  }>;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  @summary("Delete a face from a person in person directory by specified personId and persistedFaceId.")
+  @doc("""
+  Adding/deleting faces to/from a same person will be processed sequentially. Adding/deleting faces to/from different persons are processed in parallel.
+  """)
+  @returnsDoc(PersonModificationAccepted)
+  deletePersonFace is FaceResourceDeleteLongRunningOperation<PersonDirectoryFace>;
+
+  @doc("""
+  Retrieve person face information. The persisted person face is specified by its personId. recognitionModel, and persistedFaceId.
+  """)
+  getPersonFace is Operations.ResourceRead<PersonDirectoryFace>;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  @doc("""
+  Update a person persisted face's userData field.
+  """)
+  @updatesResource(PersonDirectoryFace)
+  updatePersonFace is FaceResourceUpdateOperation<
+    PersonDirectoryFace,
+    Foundations.ResourceBody<PersonDirectoryFace>
+  >;
+
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Doesn't fit standard ops"
+  @doc("""
+  Retrieve a person's persistedFaceIds representing the registered person face feature(s).
+  """)
+  @sharedRoute
```

Comment:
nit, there is `@sharedRoute`, but it is pretty hard for me to find what is the other operation that this op shares the endpoint.

Reply:
Removed, should be some workaround for previous version compiler, it was complaining for same path but different HTTP method operations.